src/printf/printf.c

```diff
@@ -624,7 +624,10 @@ static double apply_scaling(double num, struct scaling_factor normalization)
 
 static double unapply_scaling(double normalized, struct scaling_factor normalization)
 {
+#pragma GCC diagnostic push
```

Comment:
You need:
```
#if defined(__GNUC__)
```
since this is a GCC-specific pragma (clang has it but doesn't have the problem).